### ISSUE TITLE
Infinite Depth Issue

### ISSUE THREAD

**Issue by feipan664** [bolianchen/pytorch_depth_from_videos_in_the_wild#9]
Hi Bolian, after my training using your code, I tested my model and found there is infinite depth resulting from the moving objects.
![Snipaste_2022-05-01_21-04-03](https://user-images.githubusercontent.com/18341349/166145102-1684e5a7-1ec0-407f-b50d-e6584cda5c2f.png)

I wonder if this infinite depth as "holes" often appears in your experiments and how to alleviate such an issue.

Thanks.

**Comment by bolianchen** [comment]
Dear @feipan664 
Thanks very much for pointing out this problem. We have been noticed it since the official TensorFlow repository. It is an unsolved issue, so I would like to highlight this issue as "help wanted" to encourage anyone who might consider using this repository to work together.  Soon I am going to share our experiences and thoughts about what measures did alleviate it (more like workrounds) and what measures may possibly alleviate it.

**Comment by bolianchen** [comment]
The "blackholes" issue is an intrinsic problem for unsupervised depth estimation methods whose main supervision is based on image reconstruction. While objects having the same velocity with the camera, they seems not changing in the adjacent frames and behaves like faraway things. Then infinity is an easy solution for the depth network to reconstruct images properly. ["Depth from videos in the wild"](https://arxiv.org/abs/1904.04998) does impose additional degrees of freedom to counteract the issue, but unfortunately its depth prediction still has the same tendency. I am tweaking the loss components to see if I can find any effective way to deal with the issue.

**Comment by bolianchen** [comment]
Dear @feipan664 
It probably would take longer to tweak losses and do experiments to check their effects. I would like to share the workaround we found. You may try to remove all the lines containing "2011_09_29_drive_0004_sync_02" & "2011_09_29_drive_0004_sync_03" in train_files.txt. We found the "blackholes" issue trained on KITTI is mainly due to the dynamic objects in this scene.

<img width="294" alt="Screen Shot 2022-05-02 at 4 20 38 PM" src="https://user-images.githubusercontent.com/39755828/166209289-8b9da48e-73d9-4fd5-b3a0-80308fa78e9d.png">

**Comment by feipan664** [comment]
Thank you so much for the answer. I will definitely check them out. I have also read recent works and this problem looks like very hard to deal with.

**Comment by JiaoRunyu** [comment]
> Dear @feipan664 Thanks very much for pointing out this problem. We have been noticed it since the official TensorFlow repository. It is an unsolved issue, so I would like to highlight this issue as "help wanted" to encourage anyone who might consider using this repository to work together. Soon I am going to share our experiences and thoughts about what measures did alleviate it (more like workrounds) and what measures may possibly alleviate it.

Hello,dear Chen,heve you ever try this work[(struct2depth)](https://sites.google.com/view/struct2depth),they use mask_rcnn and try to modeling 3D object motion and ego-motion to solve this moving object problem,and depth_from_videos_in_wild code is also based on this,did it works?I'm trying to test it now.
